src/Telemetry/AzureDevOpsClient/AzureDevOpsClient.cs

```diff
@@ -23,22 +24,27 @@ public sealed class AzureDevOpsClient : IAzureDevOpsClient
         private readonly SemaphoreSlim _parallelism;
 
         public AzureDevOpsClient(
-            string baseUrl,
-            string organization,
-            int maxParallelRequests,
-            string accessToken)
+            AzureDevOpsClientOptions options,
+            RetryAfterHandler retryAfterHandler = null)
```

Comment:
It feels more natural for this to take an "IHttpClientFactory", and just ask it for an HttpClient, rather than having all the HTTP specifics hooked up in here.

Reply:
Done